src/services/codefixes/inferFromUsage.ts

```diff
@@ -647,6 +730,21 @@ namespace ts.codefix {
             }
         }
 
+        function inferTypeFromPropertyAssignment(assignment: PropertyAssignment | ShorthandPropertyAssignment, checker: TypeChecker, usageContext: UsageContext) {
+            const objectLiteral = isShorthandPropertyAssignment(assignment) ?
+                assignment.parent :
+                assignment.parent.parent;
+            const nodeWithRealType = isVariableDeclaration(objectLiteral.parent) ?
+                objectLiteral.parent :
+                objectLiteral;
+
+            addCandidateThisType(usageContext, checker.getTypeAtLocation(nodeWithRealType));
+        }
+
+        function inferTypeFromPropertyDeclaration(declaration: PropertyDeclaration, checker: TypeChecker, usageContext: UsageContext) {

```

Comment:
If a function is used as the initializer of a static property declaration, should the type include `typeof`?

```ts
const returnThisMember = function ([| |]) {
    return this.member;
}

class Container {
    member = "sample";
    returnThisMember = returnThisMember;
};
```

Right now it just becomes `this: Container`. If that's desirable, could you give me a pointer on how to get the `typeof` equivalent type from `declaration.parent`? I haven't figured it out 🙁

Reply:
In your example, did you mean to say `static returnThisMember = ...`? Otherwise I need more clarification.

Reply:
Sorry, yes, that's correct.

Reply:
Inside the checker it's `getTypeOfSymbol` vs `getDeclaredTypeOfSymbol` -- both types live on the same symbol. Outside the checker I think it's `getTypeAtLocation` vs `getDeclaredTypeOfSymbol`, so you'll have to get the symbol for the latter function.